sphinx/ext/napoleon/docstring.py

```diff
@@ -1181,37 +1216,20 @@ def push_item(name: str, rest: List[str]) -> None:
         if not items:
             return []
 
-        roles = {
-            'method': 'meth',
-            'meth': 'meth',
-            'function': 'func',
-            'func': 'func',
-            'class': 'class',
-            'exception': 'exc',
-            'exc': 'exc',
-            'object': 'obj',
-            'obj': 'obj',
-            'module': 'mod',
-            'mod': 'mod',
-            'data': 'data',
-            'constant': 'const',
-            'const': 'const',
-            'attribute': 'attr',
-            'attr': 'attr'
-        }
-        if self._what is None:
-            func_role = 'obj'
-        else:
-            func_role = roles.get(self._what, '')
+        # apply type aliases
+        items = [
+            translate(func, description, role)
+            for func, description, role in items
+        ]
+
+        func_role = 'obj'
         lines = []  # type: List[str]
         last_had_desc = True
         for func, desc, role in items:
             if role:
                 link = ':%s:`%s`' % (role, func)
-            elif func_role:
-                link = ':%s:`%s`' % (func_role, func)
             else:
-                link = "`%s`_" % func
+                link = ':%s:`%s`' % (func_role, func)
```

Comment:
It seems `func_role` would not be changed. How about embedding it into the format-string?